python/rmm/_lib/device_buffer.pxd

```diff
@@ -0,0 +1,26 @@
+from libcpp.memory cimport unique_ptr
+
+from rmm._lib.lib cimport cudaStream_t
+
+cdef extern from "rmm/device_buffer.hpp" namespace "rmm" nogil:
+    cdef cppclass device_buffer:
+        device_buffer()
+        device_buffer(size_t size)
+        device_buffer(const void* source_data, size_t size)
+        device_buffer(const device_buffer& other)
```

Comment:
I'm guessing we don't care about using the move constructor?

Reply:
Does Cython support the move operator?

Reply:
> Does Cython support the move operator?

Kinda... @shwina has a workaround, but it needs to be redeclared for each type you intend to call move on

See here: https://github.com/rapidsai/cudf/pull/3254/files?file-filters%5B%5D=.md&file-filters%5B%5D=.pxd&file-filters%5B%5D=.py&file-filters%5B%5D=.pyx&file-filters%5B%5D=.txt&file-filters%5B%5D=dotfile#diff-d01cda1b4984dcf102001556cd992b3aR123-R128

Reply:
I can't think of a situation in which we'll need the move constructor.

Reply:
I see. So it just tricks Cython long enough to get some useable C++ code. Interesting to see that works.

This looks related ( https://github.com/cython/cython/issues/1612 ).